tensorflow/lite/experimental/litert/vendors/qualcomm/core/builders/embedding_lookup_op_builder.cc

```diff
@@ -1,21 +1,53 @@
-// Copyright (c) Qualcomm Innovation Center, Inc.
-// All Rights Reserved.
+// Copyright (c) Qualcomm Innovation Center, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
 
 #include "tensorflow/lite/experimental/litert/vendors/qualcomm/core/builders/embedding_lookup_op_builder.h"
 
+#include <numeric>
+
 namespace qnn {
+namespace {
+constexpr int kTableIdx = 1;
+constexpr int kIndicesIdx = 0;
+constexpr int kOutputIdx = 0;
+}  // namespace
 
 std::vector<OpWrapper> BuildEmbeddingLookupOp(
     TensorPool& tensor_pool, const std::vector<TensorWrapperRef>& inputs,
     const std::vector<TensorWrapperRef>& outputs) {
   std::vector<OpWrapper> res;
 
+  TensorWrapper& table_tensor = inputs[kTableIdx];
+  TensorWrapper& indices_tensor = inputs[kIndicesIdx];
+  TensorWrapper& output_tensor = outputs[kOutputIdx];
+
   auto& gather_op = CreateOpWrapper(res, QNN_OP_GATHER);
-  gather_op.AddInputTensor(inputs[1]);
-  gather_op.AddInputTensor(inputs[0]);
-  for (const auto& output : outputs) {
-    gather_op.AddOutputTensor(output);
+  // Case: QInt8 table with QInt16 output
+  if (table_tensor.IsQuant8() && output_tensor.IsQuant16()) {
+    std::vector<std::int16_t> int16_data;
+    size_t data_len =
+        std::accumulate(table_tensor.GetDims().begin(),
+                        table_tensor.GetDims().end(), 1, std::multiplies<>());
+    // TODO: do not cast
+    auto* int8_data = reinterpret_cast<const std::int8_t*>(
+        table_tensor.GetStaticTensorData());
+    for (int i = 0; i < data_len; ++i) {
+      int16_data.emplace_back(static_cast<std::int16_t>(int8_data[i]));
```

Comment:
Nit: compiler should do this implicitly. 
```suggestion
      int16_data.emplace_back(int8_data[i]);
```